exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -223,19 +223,65 @@ public void connect(Properties props) throws RpcException {
     connect(null, props);
   }
 
+  /**
+   * Function to populate the endpointList with information of all the drillbits
+   * provided in the connection string by client
+   * @param endpointList - ArrayList of DrillbitEndpoints
+   * @param drillbits - One or more drillbit ip[:port] provided in connection string
+   */
+  public void populateEndpointsList(ArrayList<DrillbitEndpoint> endpointList, String drillbits){
+
+    // If no information about drillbits is provided then just return empty list.
+    if(drillbits == null || drillbits.length() == 0){
+      return;
+    }
+    final String[] connectInfo = drillbits.split(",");
+
+    /* For direct connection we can get URL string having drillbit property as below:
+         drillbit=<ip>:<port> --- Use the IP and port specified as the Foreman IP and port
+         drillbit=<ip>        --- Use the IP specified as the Foreman IP with default port in config file
+         drillbit=<ip1>:<port1>,<ip2>:<port2>... --- Randomly select the IP and port pair from the specified
+                                                     list as the Foreman IP and port.
+
+       Fetch ip address and port information for each drillbit and populate the list
+    */
+    for(String info : connectInfo){
+      info = info.trim();
+
+      if(info != null){
+        // Split each info to get ip address and port value
+        final String[] drillbitInfo = info.split(":");
+
+        // Check for malformed ip:port string
+        if(drillbitInfo == null || drillbitInfo.length == 0){
```

Comment:
Length can be 0 here when the string just contain ":" 

In case of more than 2 ports right now I am falling back to default port. But as discussed changed it to throw new exception "InvalidConnectionInfoException"